### ISSUE TITLE
Update ingest workflow rule order

### ISSUE THREAD

**Issue by joverlee521** [nextstrain/hepatitisB#9]
In reviewing scripts for centralizing into [nextstrain/ingest](https://github.com/nextstrain/ingest), I realized this repo has a [`tsv-to-ndjson`](https://github.com/nextstrain/hepatitisB/blob/1f38f623d493bbafc25a4ab60226a4bd59ef8f6d/ingest/scripts/tsv-to-ndjson.py) script instead of the usual `csv-to-ndjson` script. This led me to look into the [ingest workflow rules](https://github.com/nextstrain/hepatitisB/blob/1f38f623d493bbafc25a4ab60226a4bd59ef8f6d/ingest/ingest.smk) in more detail. 

I'm proposing the following to align the workflow with the general pattern of other pathogen ingests (and drop the need for the `tsv-to-ndjson` script):

1. Modify `parse-genbank` to output NDJSON records instead of metadata TSV + sequences FASTA.
2. Change the rule order and input/outputs to:
```
parse_genbank:
    input: genbank.gb
    output: genbank.ndjson

transform_metadata:
    input: genbank.ndjson
    output: 
        metadata.tsv, 
        sequence.fasta

re_circularise:
    Same input/output formats

align_everything:
    Same input/output formats 

join_nextclade_metadata: 
    Same input/output formats
```

@jameshadfield does this make sense? Or is there any particular reason the re_circularise step needs to run before the metadata clean up?

**Comment by jameshadfield** [comment]
> Or is there any particular reason the re_circularise step needs to run before the metadata clean up?

I don't think so. And `re_circularise` will be removed once we use Nextclade v3 as that functionality is now built-in I believe.

> I'm proposing the following to align the workflow with the general pattern of other pathogen ingests...

In the context of making HepB the 2nd canonical workflow (alongside the primary "general organization of the repo/workflow, without any pathogen specific bits") then this seems fine. In general, I'm not concerned if pathogen repos want to diverge in how exactly they handle individual ingest steps.

**Comment by joverlee521** [comment]
> In general, I'm not concerned if pathogen repos want to diverge in how exactly they handle individual ingest steps.

Yup, I totally get that and I would always expect there to be custom steps in individual pathogen ingests. I'm just trying to standardize as much as possible across the board so that it's easier for the team to maintain.

**Comment by jameshadfield** [comment]
Resolved by #13, using `git curate passthru` (sic) as that's replaced `csv-to-ndjson` since this issue was made!
